### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**

### CODE REVIEW

File: src/makeInfinityAdder.js [hunk 1]
```diff
@@ -6,7 +6,23 @@
...
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (num) => {
+    if (num || num === 0) {
+      amount += num;
+
+      return adder;
+    }
```
Comment (dmtrhrytsak) [inline src/makeInfinityAdder.js:16]: If I'm right, you check here whether the argument `num` was passed?

```
 if (num || num === 0) {
      amount += num;

      return adder;
    }
```

And you realized that if a user passes 0 it won't add this num, but it should) So you added another checking for that `num === 0`. But, wouldn't it be more obvious if you were checking the number of passed arguments themselves? You could do that through `arguments.length` 

One important note here is that arrow functions don't have `arguments` property, so we should use functions instead)

**Review by dmtrhrytsak — APPROVED** [review]
This script is so durable that I'm pretty sure It could add numbers non-stop for the rest of my days) Awesome!

File: src/makeInfinityAdder.js (hunk 1 above)
Comment (dariia-romanova) [reply src/makeInfinityAdder.js:16]: Yeah, I was wondering why arguments property doesn't work.
arguments.length is obviously better choice, thanks)

File: src/makeInfinityAdder.js [hunk 2]
```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = function(num) {
+    if (arguments.length !== 0) {
```
Comment (ihor-jpeg) [inline src/makeInfinityAdder.js:12]: this is correct. an alternative would be `num !== undefined`

**Review by ihor-jpeg — APPROVED** [review]
Good work!

**Review by ihor-jpeg — APPROVED** [review]

**Review by ihor-jpeg — CHANGES_REQUESTED** [review]
.

**Review by ihor-jpeg — APPROVED** [review]

**Pull request closed without merging**